samples/weather_function/src/main/kotlin/org/example/weather_func/json.kt

```diff
@@ -0,0 +1,162 @@
+	countryCode: String, 
+	windInfo: Pair<Double, Int>,
+	conditions: Array<Pair<String, String>>,
+	humidity: Int,
+	tempInfo: Triple<Int, Int, Int>
+) = Weather(
+	placeName = placeName,
+	countryCode = countryCode,
+	conditions = conditions,
+	humidity = humidity,
+	windSpeed = windInfo.first,
+	windDegrees = windInfo.second,
+	temp = tempInfo.first,
+	minTemp = tempInfo.second,
+	maxTemp = tempInfo.third
+)
+
+private fun extractTempInfo(rootObj: CPointer<cJSON>?): Triple<Int, Int, Int> {
+	val mainObj = jsonObjectItem(rootObj, "main")
+	val temp = jsonObjectItem(mainObj, "temp").intValue()
+	val minTemp = jsonObjectItem(mainObj, "temp_min").intValue()
+	val maxTemp = jsonObjectItem(mainObj, "temp_max").intValue()
+
+	return Triple(temp, minTemp, maxTemp)
+}
+
+private fun extractHumidity(rootObj: CPointer<cJSON>?): Int {
+	val mainObj = jsonObjectItem(rootObj, "main")
+
+	return jsonObjectItem(mainObj, "humidity").intValue()
+}
+
+private fun extractConditions(rootObj: CPointer<cJSON>?): Array<Pair<String, String>> {
+	val tmp = mutableListOf<Pair<String, String>>()
+	val weatherObj = jsonObjectItem(rootObj, "weather")
+
+	for (pos in 0..(jsonArraySize(weatherObj) - 1)) {
+		val item = jsonArrayItem(weatherObj, pos)
+		val name = jsonObjectItem(item, "main").stringValue()
+		val desc = jsonObjectItem(item, "description").stringValue()
+
+		tmp += (name to desc)
+	}
+	return tmp.toTypedArray()
+}
+
+private fun extractWindInfo(rootObj: CPointer<cJSON>?): Pair<Double, Int> {
+	val windObj = jsonObjectItem(rootObj, "wind")
+	val windSpeed = jsonObjectItem(windObj, "speed").doubleValue()
+	val windDegrees = jsonObjectItem(windObj, "deg").intValue()
+
+	return (windSpeed to windDegrees)
+}
+
+private fun extractCountryCode(rootObj: CPointer<cJSON>?): String {
+	val sys = jsonObjectItem(rootObj, "sys")
+	
+	return jsonObjectItem(sys, "country").stringValue()
+}
+
+internal fun weatherToJsonString(weather: Weather) : String {
+	val rootObj = createJsonObject()
+	addStringToObject(rootObj, "placeName", weather.placeName)
+	addStringToObject(rootObj, "countryCode", weather.countryCode)
+	addNumberToObject(rootObj, "humidity", weather.humidity.toDouble())
+	addTempInfoArray(rootObj, temp = weather.temp, minTemp = weather.minTemp, maxTemp = weather.maxTemp)
+	addConditionsArray(rootObj, weather.conditions)
+	val result = jsonString(rootObj)?.toKString() ?: ""
+	deleteObject(rootObj)
+	return result
+}
+
+private fun addTempInfoArray(rootObj: CPointer<cJSON>?, temp: Int, minTemp: Int, maxTemp: Int) {
+	val tmpObj = createJsonObject()
+
+	addNumberToObject(tmpObj, "temp", temp.toDouble())
+	addNumberToObject(tmpObj, "minTemp", minTemp.toDouble())
+	addNumberToObject(tmpObj, "maxTemp", maxTemp.toDouble())
+	addItemToObject(rootObj, "tempInfo", tmpObj)
+}
+
+private fun addConditionsArray(rootObj: CPointer<cJSON>?, conditions: Array<Pair<String, String>>) {
+	val tmpArr = addArrayToObject(rootObj, "conditions")
+
+	conditions.forEach { c ->
```

Comment:
what is `c`? Can be ommited and we can use `it` here...

Reply:
The **c** parameter in the lambda stands for condition which is fine in this case. Since a [multi line lambda](https://kotlinlang.org/docs/reference/coding-conventions.html#lambda-formatting) is being used it wouldn't be appropriate to use **it** as a parameter name.